### PR TITLE
test: add missing integration test for PouchContainer

### PR DESCRIPTION
**Pull request by allencloud**
Signed-off-by: Allen Sun <allensun.shl@alibaba-inc.com>

### Ⅰ. Describe what this PR did

This PR fixed the following issues:
* start an exec in a non-running container should return a status code of 409.

In `内部PouchContainer NightlyBuild 测试看板 20181224-041536`:
```

TestCase | 测试通过率 | Alios310 | Alios310 | Alios310 | Alios49 | Alios49 | Alios49 | Owner
-- | -- | -- | -- | -- | -- | -- | -- | --
PouchLogsSuite.TestLogsWithDetails | 83.3% | PASS | PASS | PASS | FAIL | PASS | PASS | zhangyue zy675793960@yeah.net
APIContainerAttachSuite.TestContainerAttachNotFound | 100.0% | SKIP | SKIP | SKIP | SKIP | SKIP | SKIP | letty letty.ll@alibaba-inc.com
APIContainerAttachSuite.TestContainerAttachStdin | 100.0% | SKIP | SKIP | SKIP | SKIP | SKIP | SKIP | letty letty.ll@alibaba-inc.com
APIContainerCheckpointSuite.TestCheckpointCreateAPI | 100.0% | SKIP | SKIP | SKIP | SKIP | SKIP | SKIP | Ace-Tang aceapril@126.com
APIContainerCheckpointSuite.TestCheckpointDelAPI | 100.0% | SKIP | SKIP | SKIP | SKIP | SKIP | SKIP | Ace-Tang aceapril@126.com
APIContainerCheckpointSuite.TestCheckpointListAPI | 100.0% | SKIP | SKIP | SKIP | SKIP | SKIP | SKIP | Ace-Tang aceapril@126.com
APIContainerCreateSuite.TestBadParam | 100.0% | SKIP | SKIP | SKIP | SKIP | SKIP | SKIP | letty letty.ll@alibaba-inc.com
APIContainerExecInspectSuite.TestContainerExecInspectOk | 100.0% | SKIP | SKIP | SKIP | SKIP | SKIP | SKIP | Allen Sun allensun.shl@alibaba-inc.com
APIContainerExecStartSuite.TestContainerExecStartDup | 100.0% | SKIP | SKIP | SKIP | SKIP | SKIP | SKIP | letty.ll letty.ll@alibaba-inc.com
APIContainerExecStartSuite.TestContainerExecStartPaused | 100.0% | SKIP | SKIP | SKIP | SKIP | SKIP | SKIP | letty.ll letty.ll@alibaba-inc.com
APIContainerExecStartSuite.TestContainerExecStartStopped | 100.0% | SKIP | SKIP | SKIP | SKIP | SKIP | SKIP | letty.ll letty.ll@alibaba-inc.com
APIContainerStartSuite.TestInvalidParam | 100.0% | SKIP | SKIP | SKIP | SKIP | SKIP | SKIP | letty letty.ll@alibaba-inc.com
PouchCheckpointSuite.TestCheckpointCreate | 100.0% | SKIP | SKIP | SKIP | SKIP | SKIP | SKIP | Ace-Tang aceapril@126.com
PouchCheckpointSuite.TestCheckpointDelete | 100.0% | SKIP | SKIP | SKIP | SKIP | SKIP | SKIP | Ace-Tang aceapril@126.com
PouchCheckpointSuite.TestCheckpointList | 100.0% | SKIP | SKIP | SKIP | SKIP | SKIP | SKIP | Ace-Tang aceapril@126.com
PouchCommitSuite.TestCommitBindMount | 100.0% | SKIP | SKIP | SKIP | SKIP | SKIP | SKIP | Ace-Tang aceapril@126.com
PouchCommitSuite.TestCommitContainer | 100.0% | SKIP | SKIP | SKIP | SKIP | SKIP | SKIP | Ace-Tang aceapril@126.com
PouchCommitSuite.TestCommitHardLink | 100.0% | SKIP | SKIP | SKIP | SKIP | SKIP | SKIP | Ace-Tang aceapril@126.com
PouchCommitSuite.TestCommitNewFile | 100.0% | SKIP | SKIP | SKIP | SKIP | SKIP | SKIP | Ace-Tang aceapril@126.com
PouchNetworkSuite.TestNetworkConnectWithRestart | 100.0% | SKIP | SKIP | SKIP | SKIP | SKIP | SKIP | Eric Li lcy041536@gmail.com
PouchRichContainerSuite.TestRichContainerSystemdWorks | 100.0% | SKIP | SKIP | SKIP | SKIP | SKIP | SKIP | letty letty.ll@alibaba-inc.com
PouchRunBlkioSuite.TestRunBlockIOWeightDevice | 100.0% | SKIP | SKIP | SKIP | SKIP | SKIP | SKIP | letty letty.ll@alibaba-inc.com
PouchRunDeviceSuite.TestRunDeviceDirectory | 100.0% | SKIP | SKIP | SKIP | SKIP | SKIP | SKIP | HusterWan zirenwan@gmail.com
PouchSnapshotterSuite.TestNotSetSnapshotter | 100.0% | SKIP | SKIP | SKIP | SKIP | SKIP | SKIP | NA
PouchSnapshotterSuite.TestOldSnapshotterNotClean | 100.0% | SKIP | SKIP | SKIP | SKIP | SKIP | SKIP | NA
PouchSnapshotterSuite.TestSetDefaultSnapshotter | 100.0% | SKIP | SKIP | SKIP | SKIP | SKIP | SKIP | NA
PouchStartSuite.TestStartFromCheckpoint | 100.0% | SKIP | SKIP | SKIP | SKIP | SKIP | SKIP | Ace-Tang aceapril@126.com
```
We skipped lots of test cases.

This PR tries to add some. And could you help to try to make the rest ones to move on(Whether to continue developing or skipping).

### Ⅱ. Does this pull request fix one issue?

no

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)
added

### Ⅳ. Describe how to verify it
none

### Ⅴ. Special notes for reviews

@sunyuan3 @chuanchang

### CODE REVIEW

**Comment by chuanchang** [comment]
@allencloud it's failed to run make check, the details as follows.

```
#git rev-parse HEAD
bc9cddc3b8aadf82022d444966a835f39e4a4818

#make check
gometalinter
gometalinter --config .gometalinter.json ./...
test/api_container_exec_start_test.go:162:39:error: undefined: content (vet)
test/api_container_exec_start_test.go:164:67:error: undefined: body (vet)
test/api_container_exec_start_test.go:178:39:error: undefined: content (vet)
test/api_container_exec_start_test.go:186:12:error: no new variables on left side of := (vet)
test/api_container_exec_start_test.go:201:39:error: undefined: content (vet)
test/api_container_exec_start_test.go:202:67:error: undefined: body (vet)
test/api_container_exec_start_test.go:206:66:error: undefined: body (vet)
make: *** [gometalinter] Error 1
```

**Comment by chuanchang** [comment]
@allencloud there are 4 cases are failure in CI.

```shell
----------------------------------------------------------------------
FAIL: /home/travis/gopath/src/github.com/alibaba/pouch/test/api_container_exec_inspect_test.go:28: APIContainerExecInspectSuite.TestContainerExecInspectOk
/home/travis/gopath/src/github.com/alibaba/pouch/test/api_container_exec_inspect_test.go:80:
    c.Assert(execInspect02.Running, check.Equals, true)
... obtained bool = false
... expected bool = true
PASS: /home/travis/gopath/src/github.com/alibaba/pouch/test/api_container_exec_start_test.go:128: APIContainerExecStartSuite.TestContainerExecStart	4.109s
----------------------------------------------------------------------
FAIL: /home/travis/gopath/src/github.com/alibaba/pouch/test/api_container_exec_start_test.go:195: APIContainerExecStartSuite.TestContainerExecStartDup
/home/travis/gopath/src/github.com/alibaba/pouch/test/api_container_exec_start_test.go:213:
    CheckRespStatus(c, resp, 500)
/home/travis/gopath/src/github.com/alibaba/pouch/test/util_api.go:27:
    c.Assert(resp.StatusCode, check.Equals, status, check.Commentf("Response Body: %v", string(body)))
... obtained int = 200
... expected int = 500
... Response Body: (failed to create containerIO: conflict
----------------------------------------------------------------------
PASS: /home/travis/gopath/src/github.com/alibaba/pouch/test/api_container_exec_start_test.go:146: APIContainerExecStartSuite.TestContainerExecStartNotFound	0.003s
----------------------------------------------------------------------
FAIL: /home/travis/gopath/src/github.com/alibaba/pouch/test/api_container_exec_start_test.go:173: APIContainerExecStartSuite.TestContainerExecStartPaused
/home/travis/gopath/src/github.com/alibaba/pouch/test/api_container_exec_start_test.go:186:
    CheckRespStatus(c, resp, 200)
/home/travis/gopath/src/github.com/alibaba/pouch/test/util_api.go:27:
    c.Assert(resp.StatusCode, check.Equals, status, check.Commentf("Response Body: %v", string(body)))
... obtained int = 204
... expected int = 200
... Response Body: 
----------------------------------------------------------------------
FAIL: /home/travis/gopath/src/github.com/alibaba/pouch/test/api_container_exec_start_test.go:156: APIContainerExecStartSuite.TestContainerExecStartStopped
/home/travis/gopath/src/github.com/alibaba/pouch/test/api_container_exec_start_test.go:162:
    execid := CreateExecCmdOk(c, cname, "echo", "test")
/home/travis/gopath/src/github.com/alibaba/pouch/test/util_api.go:27:
    c.Assert(resp.StatusCode, check.Equals, status, check.Commentf("Response Body: %v", string(body)))
... obtained int = 500
... expected int = 201
... Response Body: {"message":"container af955a13ea8d00c9137c73e391b9d60413f00609e9324e6e76b0810542f190b8 is not running"}
```

**Comment by allencloud** [comment]
Unfortunately, the image pulling part fails with:
```
----------------------------------------------------------------------
FAIL: /home/travis/gopath/src/github.com/alibaba/pouch/test/z_cli_daemon_test.go:219: PouchDaemonSuite.TestDaemonRestart
/home/travis/gopath/src/github.com/alibaba/pouch/test/z_cli_daemon_test.go:232:
    c.Fatalf("pull image failed, err:%v", result)
... Error: pull image failed, err:
Command:  /usr/local/bin/pouch --host unix:///tmp/test/pouch/pouchd.sock pull registry.hub.docker.com/library/busybox:1.28
ExitCode: 1
Error:    exit status 1
Stdout:   registry.hub.docker.com/library/busybox:1.28: resolving
Stderr:   Error: failed to display progress: failed to pull image: registry.hub.docker.com/library/busybox:1.28 not found
----------------------------------------------------------------------
```
I think there is much possibility that this is a flaky case due to network issue. While I think we should try to add some retry logistics in it to reduce the failure possibility of outer issues.

File: test/api_container_exec_inspect_test.go [hunk 1]
```diff
@@ -26,70 +26,47 @@ func (suite *APIContainerExecInspectSuite) SetUpTest(c *check.C) {
...
-	CheckRespStatus(c, resp, 200)
-
-	var execInspect01 types.ContainerExecInspect
-	request.DecodeBody(&execInspect01, resp.Body)
+	// create an exec and get the execID
+	// and inspect the exec before exec start
 
-	c.Assert(execInspect01.Running, check.Equals, false)
-	c.Assert(execInspect01.ExitCode, check.Equals, int64(0))
+	execid := CreateExecCmdOk(c, cname, "sleep", "9")
```
[the remark this answers is not in the document]
Comment (allencloud) [reply test/api_container_exec_inspect_test.go]: I am wondering if we could keep it there currently. Still keep the original test case, and the test case could be done in the following releases.

**Comment by fuweid** [comment]
closing it because it is not active right now. please feel free to reopen it.

**Pull request closed without merging**